### PR TITLE
Feat/vertex bone attributes

### PR DESCRIPTION
**Pull request by setzer22**
## Checklist

- [x] cargo clippy reports no issues
- [x] cargo deny issues have been fixed or added to deny.toml
- [x] relevant examples/test cases run
- [x] changes added to changelog *Can do this later, once more substantial changes land*
  - [x] Add credit to yourself for each change: `Added new functionality @githubname`.

## Problems PR Solves
In preparation to supporting mesh skinning, add vertex bone index and bone weight attributes. Following the de-facto standard for this sort of thing, I used UVec4 for indices and Vec4 for weights, supporting up to 4 bones influencing every vertex.

## Related Issues
None, that I know of

### CODE REVIEW

**Comment by setzer22** [comment]
I just noticed the gltf crate parses bone indices as `u16`, so I changed the original `UVec4` to `[u16;4]` to use those instead.

File: rend3/src/managers/mesh.rs [hunk 1]
```diff
@@ -26,6 +26,10 @@ pub const VERTEX_UV_SIZE: usize = size_of::<Vec2>();
 pub const VERTEX_COLOR_SIZE: usize = size_of::<[u8; 4]>();
 /// Size of a single index.
 pub const INDEX_SIZE: usize = size_of::<u32>();
+/// Size of a joint index vector
+pub const VERTEX_JOINT_INDEX_SIZE: usize = size_of::<[u16; 4]>();
```
Comment (cwfitzgerald) [inline rend3/src/managers/mesh.rs:30]: Need to add these to `rend3-routine/src/common/vertex.rs` as well.

File: rend3/src/managers/mesh.rs [hunk 2]
```diff
@@ -59,6 +65,8 @@ impl MeshBuffers {
         rpass.set_vertex_buffer(3, self.vertex_uv0.slice(..));
         rpass.set_vertex_buffer(4, self.vertex_uv1.slice(..));
         rpass.set_vertex_buffer(5, self.vertex_color.slice(..));
+        rpass.set_vertex_buffer(6, self.vertex_joint_index.slice(..));
+        rpass.set_vertex_buffer(7, self.vertex_joint_weight.slice(..));
```
Comment (cwfitzgerald) [inline rend3/src/managers/mesh.rs]: There's a really subtle problem here, gpu mode adds a vertex attribute at the very end, so we need to go bump the index on that to 8.

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Looking good, minor things

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
Code LGTM, just need to modify and regen the shaders, then we're good

**Comment by setzer22** [comment]
I only adapted the location value for the object index, but didn't add type declarations for the joint indices / weights in GLSL as we're never going to be using them.

I also recompiled the shaders, and things seem to work.

**Review by cwfitzgerald — APPROVED** [review]
Thanks!

**Pull request merged**